Sources/App/Controllers/KarmaParser.swift

```diff
@@ -8,45 +8,58 @@
 import Vapor
 
 final class KarmaParser {
-    private let posRegex = "(<@([\\w].+?)>)[\\s]*(\\+{1,5}\\+)"
-    private let negRegex = "(<@([\\w].+?)>)[\\s]*(-{1,5}-)"
+    private let posRegex = "(<@[\\w].+?>)[\\s]*(\\+{1,5}\\+)"
+    private let negRegex = "(<@[\\w].+?>)[\\s]*(-{1,5}-)"
+    private let userRegexString = "<@([\\w\\d]+)>"
 
     let positiveRegex: NSRegularExpression!
     let negativeRegex: NSRegularExpression!
+    let userRegex: NSRegularExpression!
 
     init() {
         positiveRegex = try! NSRegularExpression(pattern: posRegex)
         negativeRegex = try! NSRegularExpression(pattern: negRegex)
+        userRegex = try! NSRegularExpression(pattern: userRegexString)
     }
 
-    func karmaMessageFrom(message: String) -> KarmaMessage? {
-        if let match = findMatch(regex: positiveRegex, message: message) {
+    func karmaMessages(from message: String) -> [KarmaMessage] {
+        if let match = findKarmaMatch(using: positiveRegex, on: message) {
             return match
         }
 
-        if let match = findMatch(regex: negativeRegex, message: message) {
+        if let match = findKarmaMatch(using: negativeRegex, on: message) {
             return match
         }
 
-        return nil
+        return []
     }
 
-    private func findMatch(regex: NSRegularExpression, message: String) -> KarmaMessage? {
+    private func findKarmaMatch(using regex: NSRegularExpression, on message: String) -> [KarmaMessage]? {
```

Comment:
`Matches` because of the collection return.

Do `nil` and an empty array represent different things?

Reply:
I should probably figure out a cleaner method, but it just allowed for the method: 

```
func karmaMessages(from message: String) -> [KarmaMessage] {
        if let match = findKarmaMatch(using: positiveRegex, on: message) {
            return match
        }

        if let match = findKarmaMatch(using: negativeRegex, on: message) {
            return match
        }

        return []
    }
```
Instead of having .isEmpty checks to see if it "worked" or not. Let me think on it and re-address because I agree [] and nil mean different things and is not clear.